lib/test/dynamic_config/test_dynamic_config_base.rb

```diff
@@ -0,0 +1,40 @@
+require_relative '../test_helper'
+
+require 'dynamic_config/dynamic_config_base'
+
+class DynamicConfigBaseTest < Minitest::Test
+  def setup
+    @mock_datastore = MiniTest::Mock.new
+    @dynamic_config = DynamicConfigBase.new(@mock_datastore)
+  end
+
+  def test_basic_setter_functionality
+    @mock_datastore.expect(:set, true, ["valid key", "first value"])
+    assert @dynamic_config.set("valid key", "first value")
+
+    assert_raises ArgumentError do
+      @dynamic_config.set(:invalid_key, "second value")
+    end
+
+    @mock_datastore.verify
+  end
+
+  def test_basic_getter_functionality
+    @mock_datastore.expect(:get, "value", ["valid key"])
+    assert_equal @dynamic_config.get("valid key", "default"), "value"
+
+    assert_raises ArgumentError do
+      @dynamic_config.get(:invalid_key)
+    end
+
+    @mock_datastore.verify
+  end
+
+  def test_get_default_logic
+    @mock_datastore.expect(:get, nil, ["nil key"])
```

Comment:
I don't understand what this one does. Does it support the attempt to lookup a value that does not exist in the mock datastore?

Reply:
ah, yeah, I can see how my thinking here is a bit unclear. By "nil key", I mean "a key that returns a nil value"; this test is confirming that `get` will return the default value rather than `nil` in this situation

Reply:
basically just testing this logic: https://github.com/code-dot-org/code-dot-org/pull/45418/files#diff-36cb321b2990a90b61e832d67f7e3438a5be304e3173f2cc3eafed2188c40d68R23-R25

Reply:
"expect" is being used both to verify the way in which dynamic config is invoking the underlying datastore and to provide a return value that is checked on line 36

Reply:
Actually - the problem is that I forgot that the 2nd argument of `.expect` is the *return* value of the mocked method 😆 
Apologies for wasting your time. :)

Reply:
lol, yeah, that's ALSO pretty nonintuitive; I had to re-learn that myself in order to write this :D